### PR TITLE
API update - mount() for mounting, patch() for patching

### PR DESCRIPTION
**Pull request by estrattonbailey**
As mentioned in #89, the current `patch()` API essentially does double duty: mounting to the DOM at a single node, and updating the tree as the application updates.

This PR aims to introduce a separate `mount()` function and updates the `patch()` function's API to support patching at any level of the tree.

```javascript
/** @jsx h */
import { h, mount, patch } from 'picodom'

const App = props => <h1>{props.title}</h1

let root = <App title='Hello world!' />

mount(document.body, root)

// after state updates
patch(root, (root = <App title='New title' />))
```

As you can see, it adds a few lines, but perhaps there is a more compact way 🤔 

As far as introducing a new method to the API, I think it's a good idea. "Mounting" and "patching" are different concerns, and a single API for both would be confusing IMO.

## TODO
If you like this direction, I'm happy to make any updates necessary to get it integrated 🎉 

The tests are failing currently, I could probably use some direction there: `parentNode` is undefined. Throughout the suite `patch` would need to be updated to `mount` and the params slightly rearranged.

Thanks!

### CODE REVIEW

**Comment by estrattonbailey** [comment]
# [Demo](https://github.com/estrattonbailey/picodom-demo)
Installs the library from my repo and mounts a tiny app and stateful component.

File: src/patch.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+export function mount(parent, node) {
+  var element = parent.appendChild(createElement(node))
 
   for (var cb; (cb = callbacks.pop()); cb()) {}
 
   return element
 }
 
+export function patch(oldNode, newNode) {
+  newNode._ref = patchElement(oldNode._ref.parentNode, oldNode._ref, oldNode, newNode)
```
Comment (JorgeBucaran) [inline src/patch.js]: @estrattonbailey Can we avoid doing this? I don't want to save stuff like this on the node. 🤔

Comment (estrattonbailey) [reply src/patch.js]: Ah, I meant to mention that above. The main issue I saw with patching subtrees was that there was no reference to the real DOM node for any given tree of virtual nodes, so `patchElement` didn't have any root node to mutate childnodes from.

Do you think there's another way to get a DOM reference for an arbitrary virtual node? In terms of memory consumption, maybe a `getter` would help. Or maybe you see an entirely different way of doing this?

**Comment by mindplay-dk** [comment]
Here's the approach I've been using:

```js
function mount(el, update) {
  let old
  
  return function() {
    patch(el, old, old = update())
  }
}
```

The function creates a closure for a target element and an update function, which is any function that returns a VNode. Here's an example with a stateless view:

https://jsfiddle.net/mindplay/tuq42u7t/

Because `update` is a function, this works equally well for views that require a `model` argument, for a functional component that requires `props` and `children`, or pretty much anything.

**Comment by jorgebucaran** [comment]
@estrattonbailey Thanks, but I went with this instead:

```jsx
import { h, patch } from "picodom"

document.body.appendChild(
  patch( h("h1", {}, "Hello") )
)
```

As you can see, the patch function now returns an element, when none is given, allowing you to append, if that's the behavior you prefer. Otherwise, if you pass an element, patchdom will attempt to patch **that** element, instead of using it as a container like in previous versions.

**Comment by estrattonbailey** [comment]
@jorgebucaran beautiful 😍

I should have renamed this PR, lol. My main goal here was was enable patching a VDOM node **at** a DOM node target. My tweak to `patch` made a new `mount` interface useful. It looks like your new version does exactly this, using similar assumptions to mine.

Can't wait to dive into this, 99% positive this is exactly what I needed. Thanks for your work!

**Pull request closed without merging**